diff --git a/lib.rs b/lib.rs
index c9543b8..10f30d2 100644
--- a/lib.rs
+++ b/lib.rs
@@ -1706,6 +1706,7 @@ pub enum SubdomainWildcards {
 
 #[derive(Clone, Eq, PartialEq, Hash, Debug, PartialOrd, Ord)]
 pub enum Host {
+  Any,
   Fqdn(FQDN),
   FqdnWithSubdomainWildcard(FQDN),
   Ip(IpAddr),
@@ -1754,6 +1755,10 @@ impl Host {
     s: &str,
     subdomain_wildcards: SubdomainWildcards,
   ) -> Result<Self, HostParseError> {
+    if matches!(subdomain_wildcards, SubdomainWildcards::Enabled) && s == "*" {
+      return Ok(Host::Any);
+    }
+
     if s.starts_with('[') && s.ends_with(']') {
       let ip_str = &s[1..s.len() - 1];
       let ip = strip_ipv6_zone_index(ip_str)
@@ -1860,6 +1866,7 @@ impl QueryDescriptor for NetDescriptor {
       return false;
     }
     match (&other.0, &self.0) {
+      (Host::Any, _) => true,
       (Host::Fqdn(a), Host::Fqdn(b)) => a == b,
       (Host::FqdnWithSubdomainWildcard(a), Host::Fqdn(b)) => {
         b.is_subdomain_of(a)
@@ -2092,6 +2099,7 @@ impl fmt::Display for NetDescriptor {
 impl fmt::Display for NetDescriptor {
   fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
     match &self.0 {
+      Host::Any => write!(f, "*"),
       Host::Fqdn(fqdn) => write!(f, "{fqdn}"),
       Host::FqdnWithSubdomainWildcard(fqdn) => write!(f, "*.{fqdn}"),
       Host::IpSubnet(ip_subnet) => write!(f, "{ip_subnet}"),
